### DISCUSSION TITLE
[ZK Stack] - Which address is responsible for processing and submitting blocks on L1

### DISCUSSION THREAD

**Discussion by sachanayush47** [zkSync-Community-Hub/zksync-developers#693] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [X] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.80.1

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

Recently, I was successfully able to deploy ZK stack with sepolia as L1.

Im getting "INSUFFICIENT FINDS" error. I think address which is responsible for processing blocks has 0 sepEth. Can I know which address is responsible for processing blocks on L1?

```
Sep 17 07:18:28 validium-testnet start.sh[3130269]: 2024-09-17T07:18:28.888263Z  INFO EthTxManager::loop_iteration: zksync_eth_sender::eth_tx_manager: Sending Blob tx 1 (nonce 0) at block 6707757 with base_fee_per_gas 0, priority_fee_per_gas 2000000000, blob_fee_per_gas Some(0)

Sep 17 07:18:29 validium-testnet start.sh[3130269]: 2024-09-17T07:18:29.066743Z  WARN EthTxManager::loop_iteration: zksync_eth_sender::eth_tx_manager: Error Sending Blob tx 1 (nonce 0) at block 6707757 with base_fee_per_gas 0, priority_fee_per_gas 2000000000, blob_fee_per_gas Some(0),error Ethereum gateway error: JSON-RPC request send_raw_transaction() failed: ErrorObject { code: ServerError(-32000), message: "INTERNAL_ERROR: insufficient funds", data: None }
```

### Expected Behavior

Blocks should be processed on L!

### Current Behavior

Blocks not being processed on L1 because of lack of funds. 

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
The address responsible for processing blocks on L1 in the ZKsync Era setup is typically the address of the `ExecutorFacet` contract. This contract handles the execution of transactions and management of state changes on the Ethereum mainnet.

Here are the details for the `ExecutorFacet` on the mainnet:
- **ExecutorFacet Address**: [0xD059478a564dF1353A54AC0D0e7Fc55A90b92246](https://etherscan.io/address/0xD059478a564dF1353A54AC0D0e7Fc55A90b92246#code)

Ensure that this address has sufficient funds to cover transaction fees on L1. If you are using Sepolia as your L1, make sure to fund the corresponding address on the Sepolia network.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by sachanayush47** [comment, marked as answer]
Chain's "operator" and "blob_operator" which are responsible for submitting batches on L1. After funding them with some sepEth. Problem has been fixed.
